gse-app/src/main/java/com/powsybl/gse/app/ProjectPane.java

```diff
@@ -680,6 +706,76 @@ private boolean ancestorsExistIn(List<? extends TreeItem<Object>> treeItems) {
         return found;
     }
 
+    private MenuItem createCopyProjectNodeItem(List<? extends TreeItem<Object>> selectedTreeItems) {
+        MenuItem copyMenuItem = GseMenuItem.createCopyMenuItem();
+        List<TreeItem<Object>> selectedItems = new ArrayList<>(selectedTreeItems);
+        copyMenuItem.setDisable(ancestorsExistIn(selectedItems) || selectedItems.contains(treeView.getRoot()));
+        copyMenuItem.setOnAction(event -> copy(selectedTreeItems));
+        return copyMenuItem;
+    }
+
+    private MenuItem createPasteProjectNodeItem(TreeItem<Object> selectedTreeItem) {
+        MenuItem pasteMenuItem = GseMenuItem.createPasteMenuItem();
+        final Clipboard clipboard = Clipboard.getSystemClipboard();
+        pasteMenuItem.disableProperty().bind(copied.not());
+        pasteMenuItem.setOnAction(event -> {
+            if (selectedTreeItem.getValue() instanceof ProjectFolder) {
+                ProjectFolder projectFolder = (ProjectFolder) selectedTreeItem.getValue();
+                if (clipboard.hasString()) {
```

Comment:
create a proper method that embed the extraction of copy information from clipboard. something like : `Optional<Pair<List<String>,String>>> getCopyInfo()` that would return optionnaly a pair with the list of nodeIds to be copied and the filesystem they belong to.